lineapy/db/utils.py

```diff
@@ -11,6 +12,38 @@
 logger = logging.getLogger(__name__)
 
 
+def parse_artifact_version(version) -> Union[int, str]:
+    """
+    Attempts to parse user-passed artifact version into a valid artifact version.
+    A valid artifact version is either
+    - postive int
+    - string "all" or "latest"
+
+    Raises ValueError on failure.
+    """
+    try:
+        # attempt to cast to either 'all' or 'latest'
+        str_casted = str(version)
+        if str_casted in ["all", "latest"]:
+            return str_casted
+    except ValueError:
+        pass
+
+    try:
+        # attempt int cast
+        float_casted = float(version)
+        int_casted = int(float_casted)
```

Comment:
The approach i prefer is to assume the user is doing the right things and catch only those issues wehre the input is not what you expect. this i think simplifies the flow of our code and looks more logical/maintainable.

if isinstance(version, str):
  if version in ["all", "latest"]:
    return version
  else:
    # do your float casting followed by int casting.
elif isinstance(version,int):
  if version > minversion:
    return version
  else:
    raise
else:
  notimplementederror